### PR TITLE
feat(omnix-cli): add path completion

### PR DESCRIPTION
**Pull request by sinrohit**
Add path completion to omnix-cli

### CODE REVIEW

File: crates/nix_rs/src/flake/url.rs [hunk 1]
```diff
@@ -9,14 +9,21 @@ use std::{
...
 ///
 /// Use `FromStr` to parse a string into a `FlakeUrl`. Or `From` or `Into` if
 /// you know the URL is valid.
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
-pub struct FlakeUrl(pub String);
+#[cfg_attr(feature = "clap", derive(Parser))]
+pub struct FlakeUrl {
+    #[cfg_attr(feature = "clap", clap(value_hint = ValueHint::AnyPath))]
+    pub url: String,
+}
```
Comment (srid) [inline crates/nix_rs/src/flake/url.rs:26]: There should be a way to obviate this change -- i.e., add value_hint for the whole structure. That or, you could just add cfg_attr to existing `pub String` attr.

Comment (rsrohitsingh682) [reply crates/nix_rs/src/flake/url.rs:26]: > add value_hint for the whole structure.
AFAIK clap does not allow this way.

Anyways these changes are not required for it to work, we have to annotate FlakeUrl where we want path completions. 
Like this 
Screenshot 2024-08-01 at 6 55 49 PM

Comment (srid) [reply crates/nix_rs/src/flake/url.rs:26]: Sure, let's do the minimal change required to get this merged. We can refactor later.

**Comment by srid** [comment]
Discussed with @rsrohitsingh682. Closing, because there's no cleaner way to do this, yet. 

Perhaps ask in https://github.com/clap-rs/clap/discussions as to how we can improve `clap` itself to make this simple.

**Pull request closed without merging**